chainer/utils/type_check.py

```diff
@@ -0,0 +1,281 @@
+import numpy
+
+from chainer import cuda
+
+
+class TypeInfo(object):
+
+    def __init__(self, name, index, shape, dtype):
+        self.index = index
+        name = '{0}[{1}]'.format(name, index)
+        self.name = name
+        self.shape = tuple(Shape(x, i, name) for i, x in enumerate(shape))
+        self.dtype = DtypeExpr(dtype, name)
+        self.ndim = Member(len(self.shape), name, 'ndim')
+
+    def is_none(self):
+        return self.dtype is None
+
+    def __str__(self):
+        return self.name
+
+
+class TypeInfoTuple(tuple):
+
+    def _set_name(self, name):
```

Comment:
We can simply write `type_info_tuple_instance.name = name` instead of making setter method.